src/components/Explore/panels/TraceViewPanelScene.tsx

```diff
@@ -23,6 +26,43 @@ export interface TracePanelState extends SceneObjectState {
 }
 
 export class TraceViewPanelScene extends SceneObjectBase<TracePanelState> {
+  private async getTraceErrorMessage(error: any, traceId: string): Promise<string> {
+    const errorMessage = error?.message || '';
+    const status = error?.status;
+
+    if (status === 404 || errorMessage.toLowerCase().includes('not found')) {
+      try {
+        // Get the datasource to check timeShiftEnabled configuration
+        // Ideally this error would be returned by the datasource, but it's not currently supported.
+        const datasourceUid = getDataSource(getTraceExplorationScene(this));
+        const datasource = await getDataSourceSrv().get(datasourceUid);
+
+        // Check if the datasource has traceQuery.timeShiftEnabled set to true
+        if (datasource && (datasource as any).traceQuery?.timeShiftEnabled) {
+          const timeRange = sceneGraph.getTimeRange(this).state.value;
+
+          // Get timeshift values from datasource configuration
+          const spanStartTimeShift = (datasource as any).spanStartTimeShift || 0;
+          const spanEndTimeShift = (datasource as any).spanEndTimeShift || 0;
```

Comment:
Also should not use `any` (it's returning 0) here as it's actually not getting the shifts at all because they are in the `traceQuery` object!